server/src/test/java/pl/kejbi/tin/SocketTest.groovy

```diff
@@ -0,0 +1,123 @@
+package pl.kejbi.tin
+
+import org.slf4j.Logger
+import pl.kejbi.tin.socket.Client
+import pl.kejbi.tin.socket.Server
+import pl.kejbi.tin.socket.builder.ServerSocketBuilder
+import pl.kejbi.tin.socket.builder.SocketBuilder
+import spock.lang.Specification
+
+class SocketTest extends Specification {
+
+    Socket socket;
+    ServerSocket serverSocket;
+
+    def createServerSocket() {
+        return new ServerSocketBuilder().createServerSocket().build()
+    }
+
+    def createClientSocket() {
+        return new SocketBuilder().createSocket().build()
+    }
+
+    def cleanup() {
+        if(socket != null) {
+            if(!socket.isClosed()) {
+                socket.close()
+            }
+            socket = null
+        }
+        if(serverSocket != null) {
+            if(!serverSocket.isClosed()) {
+                serverSocket.close()
+            }
+            serverSocket = null
+        }
+    }
+
+    def "should create server socket"() {
```

Comment:
warto czasami dać trochę bardziej rozbudowane nazwy testów. Ja np. tu bym napisał "should create server socket with correct local port" czy coś takiego, bo tu jest napisane co się dzieje (sekcja when), ale nie to czego oczekujemy od testu (sekcja then)

Reply:
ok